src/analyze/analysis_handler.go

```diff
@@ -6,85 +6,148 @@ import (
 	"BGPAlert/config"
 	"BGPAlert/shake_alert"
 	"fmt"
-	"path/filepath"
 	"sort"
 	"time"
 )
 
-var recentResultsFile, _ = filepath.Abs("static_data/recentFullResult.json")
-var outlierMessagesFile, _ = filepath.Abs("outlierMessages.csv")
-var AllResults []common.Result //global so that it can be added onto by parser and seen by dataHandler
-var maxPoints = 20             //the max number of points to be displayed on the graph;
-// make sure to divide this by the number
-//of points in each window or specify that maxPoints is the number of buckets that will be processed
+//need to create a map of results based on filter -> filtering pr
+var AllResults common.Result //global so that it can be added onto by parser and seen by dataHandler
+var maxPoints = 5            //i did not want to add this into config yet because it will conflic with jolene's pr
+
+const (
+	OPTPARAM   = 5.0
+	BLTMAD     = 1
+	SHAKEALERT = 2
+	BOTHALERTS = 3
+)
 
 // Takes in a Window, parses object into frequency counts, and then calls specified analysis functions
 //code to write the frequencies; the outliers; and the minReqs to files
-func AnalyzeBGPMessages(window common.Window, config *config.Configuration) common.Result {
+func AnalyzeBGPMessages(window common.Window, config *config.Configuration) {
 	lengthMap := makeLengthMap(window)
 	// Turn map into sorted array of frequencies by timestamp
-	sortedFrequencies := GetSortedFrequencies(lengthMap) //fix these duplicates with time stamps
+	sortedFrequencies, sortedTimestamps := GetSortedFrequencies(lengthMap) //fix these duplicates with time stamps
+	frequencies := AllResults.AllFreq
 
 	//the file names will contain all the timestamps for a given folder that was processed
-	bltOutliers, bltOutlierTimes, bltOutlierMessages := BltMadWindow(sortedFrequencies, window, 5) //add optimization to here
-	shakeAlertOutliers, shakeAlertOutlierTime, shakeAlertOutlierMessages := ShakeAlertWindow(sortedFrequencies, window)
 	fmt.Printf("Sorted Array of Frequencies: \n%+v\n", sortedFrequencies)
+	bltOutliers, bltIndexes := blt_mad.BltMad(sortedFrequencies, OPTPARAM)
+	shakeAlertOutliers, shakeAlertIndexes := shake_alert.FindOutliers(sortedFrequencies)
 	fmt.Printf("BLT MAD Outliers: \n%+v\n", bltOutliers)
 	fmt.Printf("ShakeAlert Outliers: \n%+v\n", shakeAlertOutliers)
 
-	//put all the results into the Result struct and pass write it out to a json
-	r := common.Result{
-		Prefix:     config.Prefix,
-		ASN:        config.Asn,
-		PeerIP:     config.PeerIP,
-		WindowSize: config.WindowSize,
-
-		Frequencies: sortedFrequencies, //make all of these maps and append onto them here; create them as global vars
+	//check the amount of the frequencies in results
+	if (len(AllResults.AllFreq) + 1) > maxPoints { //adding one more frequency would result in more than needed points
```

Comment:
on line 30 you defined the frequencies variable. I would suggest using that as the conditional.